control-plane/stor-port/src/types/v0/transport/pool.rs

```diff
@@ -391,3 +397,32 @@ impl UnlabelPool {
         Self { pool_id, label_key }
     }
 }
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+pub struct Encryption {
+    /// Cipher to be used.
+    pub cipher: Cipher,
+    /// The encryption key.
+    pub key: Option<EncryptionKey>,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+pub struct EncryptionKey {
+    /// Name of the key.
+    pub key_name: String,
+    /// The AES encryption key.
+    pub key: Vec<u8>,
+    /// AES Key length.
+    pub key_length: u32,
+    /// key2 (required for AES_XTS).
+    pub key2: Option<Vec<u8>>,
+    /// The length of key2. Must be same as key_length.
+    pub key2_length: Option<u32>,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+pub enum Cipher {
+    #[default]
+    AesCbc = 0,
+    AesXts = 1,
+}
```

Comment:
This should not be needed here, only protobuf types should convert to i32

```suggestion
    AesCbc,
    AesXts,
}
```

Reply:
Oops, of course. Thanks.